src/newline_lexer.jl

```diff
@@ -0,0 +1,375 @@
+module NewlineLexers
+
+using SIMD, ScanByte, Libdl
+
+export Lexer, find_newlines!, possibly_not_in_string
+# const compat
+macro constfield(ex)
+    @static if VERSION < v"1.8"
+        return esc(ex)
+    else
+        return Expr(:const, esc(ex))
+    end
+end
+
+@generated function __icmp_eq_u64(x::SIMD.Intrinsics.LVec{64, T}, y::SIMD.Intrinsics.LVec{64, T}) where {T <: SIMD.Intrinsics.IntegerTypes}
+    s = """
+    %res = icmp eq <64 x $(SIMD.Intrinsics.d[T])> %0, %1
+    %resb = bitcast <64 x i1> %res to i64
+    ret i64 %resb
+    """
+    return :(
+        $(Expr(:meta, :inline));
+        Base.llvmcall($s, UInt64, Tuple{SIMD.Intrinsics.LVec{64, T}, SIMD.Intrinsics.LVec{64, T}}, x, y)
+    )
+end
+@inline function _icmp_eq_u64(x::Vec{64, T}, y::Vec{64, T}) where {T <: SIMD.Intrinsics.IntegerTypes}
+    __icmp_eq_u64(x.data, y.data)
+end
+let # Feature detection -- copied from ScanByte.jl
+    llvmpath = if VERSION ≥ v"1.6.0-DEV.1429"
+        Base.libllvm_path()
+    else
+        only(filter(lib->occursin(r"LLVM\b", basename(lib)), Libdl.dllist()))
+    end
+    libllvm = Libdl.dlopen(llvmpath)
+    gethostcpufeatures = Libdl.dlsym(libllvm, :LLVMGetHostCPUFeatures)
+    features_cstring = ccall(gethostcpufeatures, Cstring, ())
+    features = split(unsafe_string(features_cstring), ',')
+    Libc.free(features_cstring)
+
+    @eval if !any(x->occursin("clmul", x), $(features))
```

Comment:
```suggestion
    @eval if !Base.BinaryPlatforms.CPUID.test_cpu_feature(Base.BinaryPlatforms.CPUID.JL_X86_vpclmulqdq)
```
I think we can replace all this business with the above; this comes from a suggestion from @gbaraldi [here](https://discourse.julialang.org/t/adding-hwloc-as-a-julia-dependency/89664/4).